cmd/resin-provision/main.go

```diff
@@ -4,139 +4,167 @@ import (
+		} else if password == "" {
+			return errors.New("Password is required")
 		}
 	}
+
+	return nil
 }
 
-func signup() (token string, err error) {
-	fmt.Println("Creating new user...")
-	if email, e := prompt(nil, "email: "); err != nil {
-		return "", e
+func oneshot(cmd *cobra.Command, args []string) error {
+	if status, err := provisioner.Status(configPath); err != nil {
+		return err
+	} else if status {
+		return nil
+	} else if token, err := getToken(); err != nil {
+		return err
+	} else if token == "" {
+		return errors.New("Wrong email or password, please try again")
+	} else if appId, err := getApp(token, application); err != nil {
+		return err
+	} else if userId, err := resin.GetUserId(token); err != nil {
+		return err
+	} else if userName, err := resin.GetUserName(token); err != nil {
+		return err
+	} else if apiKey, err := resin.GetApiKey("https://api."+domain, appId, token); err != nil {
+		return err
 	} else {
-		for {
-			fmt.Printf("password: ")
-			if p, e := gopass.GetPasswdMasked(); e != nil {
-				return "", e
-			} else {
-				fmt.Printf("confirm password: ")
-				if c, e := gopass.GetPasswdMasked(); e != nil {
-					return "", e
-				} else {
-					password := string(p)
-					confirm := string(c)
-					if password == confirm {
-						token, err = resin.Signup("https://api."+domain, email, password)
-						if err == nil && token == "" {
-							return "", errors.New("Signup failed")
-						} else {
-							return token, nil
-						}
-					} else {
-						fmt.Println("Passwords don't match, please try again.")
-					}
-				}
-			}
-		}
+		return provisioner.Provision(appId, apiKey, userId, userName, configPath, domain, dryRun)
 	}
 }
 
-func authenticate() (token string, err error) {
-	fmt.Println("Welcome to resin.io")
-	fmt.Printf(`Please log in or sign up:
-	1) Log in
-	2) Sign up
-`)
-	if input, e := prompt([]string{"1", "2"}, "> "); err != nil {
-		return "", e
+func interactive(cmd *cobra.Command, args []string) error {
+	if status, err := provisioner.Status(configPath); err != nil {
+		return err
+	} else if status {
+		return nil
+	} else if token, err := authenticate(); err != nil {
+		return err
+	} else if appId, err := getOrCreateApp(token); err != nil {
+		return err
+	} else if userId, err := resin.GetUserId(token); err != nil {
+		return err
+	} else if userName, err := resin.GetUserName(token); err != nil {
+		return err
+	} else if apiKey, err := resin.GetApiKey("https://api."+domain, appId, token); err != nil {
+		return err
 	} else {
-		switch input {
-		case "1":
-			return login()
-		case "2":
-			return signup()
-		}
+		return provisioner.Provision(appId, apiKey, userId, userName, configPath, domain, dryRun)
 	}
+}
 
-	return "", nil
+func success(cmd *cobra.Command, args []string) error {
+	fmt.Println("Your device is now provisioned and is " +
+		"downloading and installing the resin supervisor")
+	fmt.Println("Your device will show as configuring during " +
```

Comment:
This might not happen if the provisioner doesn't provision the device, it will probably go straight to "online"